File: src/Physics/fem.cc
#include "physics.hh"
#include "../Mesh/femesh.hh"
#include <iostream>

template <int dim>
class FEM : public Physics<dim> {
protected:
    Mesh::FEMesh<dim>* mesh;
    double dtmin;
public:
    using Vector = Lin::Vector<dim>;
    using VectorField = Field<Vector>;
    using ScalarField = Field<double>;

    FEM(NodeList* nodeList, PhysicalConstants& constants, Mesh::FEMesh<dim>* mesh) : 
        Physics<dim>(nodeList,constants),
        mesh(mesh) {
    }

    ~FEM() { delete mesh; }
};

File: src/Physics/fem.py
from PYB11Generator import *
from physics import *

@PYB11template("dim")
class FEM(Physics):
    def pyinit(self,
               nodeList="NodeList*",
               constants="PhysicalConstants&",
               mesh="Mesh::FEMesh<%(dim)s>*"):
        return

FEM1d = PYB11TemplateClass(FEM,
                              template_parameters = ("1"),
                              cppname = "FEM<1>",
                              pyname = "FEM1d",
                              docext = " (1D).")
FEM2d = PYB11TemplateClass(FEM,
                              template_parameters = ("2"),
                              cppname = "FEM<2>",
                              pyname = "FEM2d",
                              docext = " (2D).")
FEM3d = PYB11TemplateClass(FEM,
                              template_parameters = ("3"),
                              cppname = "FEM<3>",
                              pyname = "FEM3d",
                              docext = " (3D).") 